front/components/assistant/conversation/ConversationParticipants.tsx

```diff
@@ -1,59 +1,36 @@
 import { Avatar } from "@dust-tt/sparkle";
-import type { ConversationType } from "@dust-tt/types";
+import type { WorkspaceType } from "@dust-tt/types";
 import React from "react";
 
+import { useConversationParticipants } from "@app/lib/swr";
+
+interface ConversationParticipantsProps {
+  conversationId: string;
+  owner: WorkspaceType;
+}
+
 export function ConversationParticipants({
-  conversation,
-}: {
-  conversation: ConversationType;
-}) {
-  type UserParticipant = {
-    username: string;
-    fullName: string | null;
-    pictureUrl: string | null;
-  };
-  type AgentParticipant = {
-    configurationId: string;
-    name: string;
-    pictureUrl: string;
-  };
-  const userParticipantsMap = new Map<string, UserParticipant>();
-  const agentParticipantsMap = new Map<string, AgentParticipant>();
-  conversation.content.map((messages) => {
-    messages.map((m) => {
-      if (m.type === "user_message") {
-        const key = `${m.context.username}-${m.context.profilePictureUrl}`;
-        if (!userParticipantsMap.has(key)) {
-          userParticipantsMap.set(key, {
-            username: m.context.username,
-            fullName: m.context.fullName,
-            pictureUrl: m.user?.image || m.context.profilePictureUrl,
-          });
-        }
-      } else if (m.type === "agent_message") {
-        const key = `${m.configuration.sId}`;
-        if (!agentParticipantsMap.has(key)) {
-          agentParticipantsMap.set(key, {
-            configurationId: m.configuration.sId,
-            name: m.configuration.name,
-            pictureUrl: m.configuration.pictureUrl,
-          });
-        }
-      }
-    });
+  conversationId,
+  owner,
+}: ConversationParticipantsProps) {
+  const { conversationParticipants } = useConversationParticipants({
+    conversationId,
+    workspaceId: owner.sId,
   });
-  const userParticipants = Array.from(userParticipantsMap.values());
-  const agentParticipants = Array.from(agentParticipantsMap.values());
+
+  if (!conversationParticipants) {
+    return <></>;
+  }
+
+  const { agents, users } = conversationParticipants;
 
   return (
     <div className="flex gap-6">
       <Avatar.Stack
         size="sm"
-        nbMoreItems={
-          agentParticipants.length > 4 ? agentParticipants.length - 4 : 0
-        }
+        nbMoreItems={agents.length > 4 ? agents.length - 4 : 0}
```

Comment:
nit: is this the same as `Math.max(agents.length - 4, 0)` ?

Reply:
Yep, will refactor.